libc/calls/ioctl.c

```diff
@@ -106,7 +106,8 @@ static int ioctl_fionread(int fd, uint32_t *arg) {
       int bytes = CountConsoleInputBytes();
       *arg = MAX(0, bytes);
       return 0;
-    } else if (g_fds.p[fd].kind == kFdDevNull) {
+    } else if (g_fds.p[fd].kind == kFdDevNull ||
+               g_fds.p[fd].kind == kFdDevRandom) {
       *arg = 1;
```

Comment:
Is that true in this case? Presumably a read call will give you some number of bytes without blocking, would it make sense to put some number here? Like I dunno, 128 or 256? (If `ProcessPrng` tops out at some number, that's probably the one to use. Otherwise `getrandom` stops at 256, so maybe that?)

If this is what `urandom` does on other systems I'm fine with keeping it.

Reply:
I tried it out, it wasn't true even for `/dev/null`. Updated to Linux behavior: return `enotty()`  on `/dev/null` and `einval()` for `/dev/(u)random`.

Reply:
Thank you for digging into that.